app/mixins/search_resource.js

```diff
@@ -43,6 +43,7 @@ export default Ember.Mixin.create({
   onSearchTextChange: Ember.observer("searchText", function() {
     if (this.isValidTextLength()) {
       this.reloadResults();
+      this.get("cache").clear();
```

Comment:
you should probably move it before the `reloadResults`

Reply:
actually, best to put it *inside*  `reloadResults()`

Reply:
and actually, you can't use `this.get("cache")` here. Because it's declared in the controller, but not the mixin. The mixin cannot _assume_ that there is a cache

Reply:
so in the _controller_

You can do

```
reloadResults() {
   this.get("cache").clear();
   this._super();
}
```